standardtracer/propagation.go

```diff
@@ -177,52 +181,48 @@ func (p *splitBinaryPropagator) JoinTrace(
 	var traceID, propagatedSpanID int64
 	var sampledByte byte
 
-	err = binary.Read(contextReader, binary.BigEndian, &traceID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &traceID); err != nil {
 		return nil, opentracing.ErrTraceCorrupted
 	}
-	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &propagatedSpanID); err != nil {
```

Comment:
the original plan for `standardtracer` was to serve as an abstract base for real tracers, which would be responsible for providing their own trace context serialization and reporter. As you can see, there is a lot of logic in the standardtracer that deals with the main API implementation without knowing how the context is serialized or spans are reported. Essentially, it was a Service Provider Interface.

But considering how many disagreements we went through about the main OpenTracing API, maintaining a shared, reusable base implementation was a bit too much at that point.

I think at this time we just want to treat it as a reference implementation, and as such it does need to make assumption about trace context serialized format. Although it doesn't mean it cannot be made easily swappable should people decide to base custom implementations on this reference implementation.